### PR TITLE
[LegacyFilters] design updates

### PR DESCRIPTION
**Pull request by jesstelford**

### CODE REVIEW

File: polaris-react/src/components/Tag/Tag.scss [hunk 1]
```diff
@@ -105,6 +105,10 @@ $button-size: 20px;
     min-height: $height;
     border-radius: var(--p-border-radius-2);
     padding-inline: calc(var(--p-space-1) + var(--p-space-05));
+
+    &.removable {
+      padding-inline-end: 0;
```
Comment (jesstelford) [inline polaris-react/src/components/Tag/Tag.scss:110]: Sneaky fix here; The `se23` `padding-inline` above accidentally overwrote the `.removable` style. Due to specificity, we have to duplicate it here so it's applied correctly.

File: polaris-react/src/components/LegacyFilters/components/ConnectedFilterControl/ConnectedFilterControl.scss [hunk 2]
```diff
@@ -59,6 +60,11 @@
 .RightContainer {
   display: flex;
 
+  /* Stop children of the filters from being overlapped due to buttons having a
+  * default min-width: 0.
+  * See more: https://stackoverflow.com/a/36247448 */
+  flex-shrink: 0;
```
Comment (jesstelford) [inline polaris-react/src/components/LegacyFilters/components/ConnectedFilterControl/ConnectedFilterControl.scss:66]: Only happens when the `children` prop is passed to `LegacyFilters`:

overlap

Disallowing shrinking fixes it:

no-overlap

File: polaris-react/src/components/LegacyFilters/LegacyFilters.stories.tsx [hunk 3]
```diff
@@ -149,7 +150,7 @@ export function WithAResourceList() {
 
   return (
     <div style={{height: '568px'}}>
-      <LegacyCard>
+      <Card padding="0">
```
Comment (jesstelford) [inline polaris-react/src/components/LegacyFilters/LegacyFilters.stories.tsx:153]: 'tis legacy, so... bu-bye.

But also, it was causing some rendering artefacts that aren't present with the new `Card`. These are just stories, so very little risk here.

**Review by sarahill — COMMENTED** [review]
1. Padding should be `12px` all around
image

2. Update buttons to use the `large` size so they match the height of the input -> we JUST added these
image

**Review by sarahill — CHANGES_REQUESTED** [review]
See [comment](https://github.com/Shopify/polaris/pull/9393#pullrequestreview-1470502704) for details

**Comment by jesstelford** [comment]
> Padding should be 12px all around

😩 This spacing has changed in the design since I opened this PR.

I'm going to close this and combine it with https://github.com/Shopify/polaris/pull/9394 into https://github.com/Shopify/polaris/pull/9376 since they're all intricately linked together (and all their designs changed since I started work on it).

**Pull request closed without merging**